flytestdlib/cache/auto_refresh.go

```diff
@@ -359,6 +359,20 @@ func (w *autoRefresh) sync(ctx context.Context) (err error) {
 	}
 }
 
+// Checks if the item is currently being processed and returns false if the item has been in processing for too long
+func (w *autoRefresh) inProcessing(key interface{}) bool {
+	item, found := w.processing.Load(key)
+	if found {
+		// handle potential race conditions where the item is in processing but not in the workqueue
+		if timeItem, ok := item.(time.Time); ok && time.Since(timeItem) > (w.syncPeriod*10) {
```

Comment:
lean to not make this configurable

Reply:
Should we increase the timeout (>30s) here? Since we enqueue batch items every 30 seconds by default.

Reply:
`w.syncPeriod*10` => 300 seconds by default. I feel we can reduce the timeout a bit. Maybe 5. wdyt